### DISCUSSION TITLE
how is this different from gorhom's bottom sheet?

### DISCUSSION THREAD

**Discussion by bryanprimus** [lodev09/react-native-true-sheet#4] [General]
~Well, I believe that this question belongs in the discussion section. But I see that we don't have that.~

However, I'm curious. What is the motivation for building this library? Is it going to be more performant than other sheet libs especially [gorhom bottom sheet](https://github.com/gorhom/react-native-bottom-sheet) since all the code implementation is on the native side? What's the direction towards? Why there is a plan to integrate with reanimated? [gorhom bottom sheet](https://github.com/gorhom/react-native-bottom-sheet) already uses reanimated btw

I'm heavily using [gorhom bottom sheet](https://github.com/gorhom/react-native-bottom-sheet) in most of my projects. The DX is good, but there are just too many little bugs that aren't handled, which I need to live with. This is probably because the library wants to achieve so many things.

**Comment by lodev09** [comment]
>  What is the motivation for building this library?

1. I previously used gorhom's bottom sheet on the App I'm working. It worked great for a bit. However, it glitches when you have lots of content (I noticed in android using FlashList). The codebase is kinda bloated already and is doing too many things for just simple bottom sheet.
2. I wanted that native iOS shrinking effect.
3. I wanted that extra blur functionality.

> Is it going to be more performant than other sheet libs especially gorhom bottom sheet since all the code implementation is on the native side

Yes, theoretically. 

> What's the direction towards?

At least for version 1, need to support inline sheet and context API. I'm thinking of similar approach with how react-navigation manages the screens. Basically the ability to present/dismiss the sheets from anywhere without passing the `ref`.

> Why there is a plan to integrate with reanimated?

![2024-04-19 20 55 56](https://github.com/lodev09/react-native-true-sheet/assets/6686328/e1f8928a-4bfd-473e-98c4-b973102a3c0f)

This is the home screen of the app I'm working on. I wanted to use the sheet here.  Currently, it's a custom reanimated/gestured view.

In order to do that, the lib needs to support `worklet` during slide/drag and also an `inline` prop.

**Reply by bryanprimus** [reply]
i agree with Gorhom's bottom sheet; it often glitches.

by the way, have you looked into this approach before? https://rnas.vercel.app/guides/sheetmanager

it helps save a lot of dev time and reduce code repetition when using bottom sheets, but it might require a provider to manage the IDs though. maybe we can introduce this approach as an option for user to use?

**Reply by lodev09** [reply]
Yeah, I was thinking with something similar. However, it doesn't really play well with the native setup. Like this [issue](https://github.com/lodev09/react-native-true-sheet/issues/5). The sheets need to be rendered along side the same view controller and cannot be placed in one place.

I can, however, introduced a `name` prop that maps into the ref handle tag and the user can just call a static method like so:
```tsx
<TrueSheet name="my-sheet">
  // ...
</TrueSheet>
```
```ts
TrueSheet.present('my-sheet')
```

I think this is much simpler 🤔

Regarding the `payload` functionality, I think it's too much for this simple package. I guess a separate package that should handle all this in the future 😅

**Reply by lodev09** [reply]
Added the `name` feature here https://github.com/lodev09/react-native-true-sheet/pull/6.

```tsx
// Define the sheet as usual
<TrueSheet name="my-sheet">
  // ...
</TrueSheet>
```
```tsx
// Somewhere in your screen
const presentMySheet = async () => {
  await TrueSheet.present('my-sheet')
  console.log('🎉')
}

return (
  <Button onPress={presentMySheet} />
)
```
